### PR TITLE
Add helm adapter

### PR DESCRIPTION
**Pull request by Sammers21**
_Closes #188

### CODE REVIEW

**Comment by Sammers21** [comment]
@g4s8, pay attention to the Pull Request.

**Comment by 0crat** [comment]
This pull request [#189](https://github.com/artipie/artipie/issues/189) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/411a3c0e-5bb6-4f06-9f0b-3956dfa0368d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/artipie/SliceFromConfig.java [hunk 1]
```diff
@@ -120,6 +121,10 @@ public SliceFromConfig(final RepoConfig config, final Vertx vertx,
                         "gem",
                         config -> CompletableFuture.completedStage(new GemSlice(storage, vertx.fileSystem()))
                     ),
+                    new MapEntry<>(
+                        "helm",
+                        config -> CompletableFuture.completedStage(new HelmSlice(storage))
```
Comment (g4s8) [inline src/main/java/com/artipie/SliceFromConfig.java:126]: @Sammers21 do you need full path URIs for Helm? If not, please wrap it with `TrimPathSlice` as some other endpoints

Comment (Sammers21) [reply src/main/java/com/artipie/SliceFromConfig.java:126]: I need the full path

**Comment by Sammers21** [comment]
> Also, please add README documentation how to use Helm adapter.

@g4s8 , I can do it in a separated pull request

**Comment by g4s8** [comment]
> > Also, please add README documentation how to use Helm adapter.
> 
> @g4s8 , I can do it in a separated pull request

@Sammers21 no, this PR is small enough and it should be done in "one transaction", since Helm adapter without documentation doesn't make any sense (and vice versa), e.g. see https://github.com/artipie/artipie/pull/190

**Comment by Sammers21** [comment]
> > > Also, please add README documentation how to use Helm adapter.
> > 
> > 
> > @g4s8 , I can do it in a separated pull request
> 
> @Sammers21 no, this PR is small enough and it should be done in "one transaction", since Helm adapter without documentation doesn't make any sense (and vice versa), e.g. see #190

@g4s8, why? The smaller changes - the better. Ain't it yours words?

**Comment by Sammers21** [comment]
@g4s8, ping, any updates?

**Comment by g4s8** [comment]
> @g4s8, ping, any updates?

@Sammers21 you still didn't update the readme.

**Comment by Sammers21** [comment]
> > @g4s8, ping, any updates?
> 
> @Sammers21 you still didn't update the readme.

I will do that in a separated pull request when the module will be ready to use.

**Comment by Sammers21** [comment]
@g4s8, I can create a puzzle if you like it better

**Comment by g4s8** [comment]
> > > @g4s8, ping, any updates?
> > 
> > 
> > @Sammers21 you still didn't update the readme.
> 
> @g4s8, I will do that in a separated pull request when the module will be ready to use.

@Sammers21 I suppose it's ready for use, if you submitted this PR. And it's a good practice to provide documentation for new functionality you're introducing. It's like you submit a new class without javadoc and say that you're going to submit javadoc in separate PR.

**Comment by Sammers21** [comment]
@g4s8, pay attention to the Pull Request.

File: README.md [hunk 2]
```diff
@@ -371,6 +371,31 @@ Install a gem:
...
+  type: helm
+  storage:
+    type: fs
+    path: /tmp/artipie/data/helm-charts
+```
+
+Publish a chart:
+
+```bash
+$ curl --data-binary "@my_chart-1.6.4.tgz" http://localhost:8080/helm
```
Comment (g4s8) [inline README.md:389]: @Sammers21 helm adapter supports "helm push plugin" https://github.com/chartmuseum/helm-push see: https://github.com/artipie/helm-adapter/issues/12
So it should be `helm repo add artipie http://localhost:8080 && helm push artipie @my_chart-1.6.4.tgz`, isn't it?

Comment (Sammers21) [reply README.md:389]: @g4s8, helm adapter is not tested with push plugin as per now and it is not as popular. I think we can add it later

Comment (g4s8) [reply README.md:389]: @Sammers21 ok, let's add it later

**Comment by g4s8** [comment]
@Sammers21 please resolve merge conflicts and we can merge

**Comment by 0crat** [comment]
Job was finished in 46 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**